config.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+
+import json
+import glob
+import re
+import argparse
+import sys
+from pathlib import Path
+
+MODULES_DIR = "/usr/share/wb-hwconf-manager/modules"
+SLOTS_PATH = "/var/lib/wb-hwconf-manager/system.conf"
+CONFIG_PATH = "/etc/wb-hardware.conf"
+
+# board slots structure
+# {
+#     "slots": [
+#         {
+#             "slot_id": "mod1",
+#             "id": "wb67-mod1",
+#             "compatible": ["wbe2", "wbe3-reduced"],
+#             "name": "Internal slot 1",
+#         },
+#         ...,
+#         {
+#               "slot_id": "extio1",
+#               "id": "wb6-extio1",
+#               "compatible": ["wb5-extio"],
+#               "name": "External I/O module 1",
+#         },
+#         ...
+#     ]
+# }
+
+# config structure
+# {
+#     "mod1": {
+#         "module": "MODULE_NAME",
+#         "options": {}
+#     },
+#     ...,
+#     "extio1": {
+#         "module": "MODULE_NAME",
+#         "options": {}
+#     },
+#     ...
+# }
+
+
+def to_old_config(config_str: str, board_slots_path: str):
+    config = json.loads(config_str)
+    # Config has slots property, it is actually old config format, pass as is
+    if "slots" in config:
+        return config
+
+    with open(board_slots_path, "r", encoding="utf-8") as board_slots_file:
+        board_slots = json.load(board_slots_file)
```

Comment:
```suggestion
        board_slots = json.load(board_slots_file)

```

а то в одну простыню слилось, подумал, что for с неправильным отступом

Reply:
Добавил пустую строку